### PR TITLE
Add Reset Password Feature

### PR DESCRIPTION
**Pull request by anirudhprabhakaran3**
# Description

Adds the password reset feature.

Fixes https://github.com/IEEE-NITK/corpus/issues/112

## Dependencies

None

## Type of Change

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [X] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration. (for bug fix / breaking change)
_Put an `x` in the boxes that apply_

- [ ] Test A
- [ ] Test B

---

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: corpus/accounts/forms.py [hunk 1]
```diff
@@ -61,3 +62,12 @@ def clean_username(self):
...
         return username
+
+
+class CorpusPasswordResetForm(PasswordResetForm):
+    def __init__(self, *args, **kwargs):
+        super(CorpusPasswordResetForm, self).__init__()
+        for visible in self.visible_fields():
+            visible.field.widget.attrs[
+                "class"
+            ] = "mt-1 block w-full rounded-md border-base-800 text-black shadow-sm focus:border-primary focus:ring focus:ring-primary-200 focus:ring-opacity-50"  # noqa: E501
```
Comment (nishant-nayak) [inline corpus/accounts/forms.py:73]: Remind me to fix this in my Achievements page PR. I've made changes to the form fields in that PR, will need to update this form accordingly.

Comment (nishant-nayak) [reply corpus/accounts/forms.py:73]: Actually, where is this being used? I don't see any reference to this Form instance. Also, the initial forgot password page does not have any styling classes applied to the input fields:

image

**Review by nishant-nayak — CHANGES_REQUESTED** [review]

File: corpus/accounts/forms.py (hunk 1 above)
Comment (anirudhprabhakaran3) [reply corpus/accounts/forms.py:73]: Yeah, I think I should remove that. When I do subclass it to add our styles, idk why but the form functionality doesn't work. For now, I'll remove it, will debug properly and figure out how to do this later.

Comment (nishant-nayak) [reply corpus/accounts/forms.py]: Can you create an issue in the backlog for this?

File: corpus/templates/registration/password_reset_confirm.html [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                            <div class="my-1">
+                                <div role="alert" class="alert alert-error">
+                                    {{ form.new_password2.errors }}
+                                </div>
+                            </div>
+                        {% endif %}
+                    </div>
+
+                    <div class="card-actions mt-5">
+                        <button type="submit" class="btn btn-primary btn-block">Send me a reset link</button>
```
Comment (nishant-nayak) [inline corpus/templates/registration/password_reset_confirm.html:51]: Update the button text. This button will reset their password, not send them another reset link.

**Review by nishant-nayak — CHANGES_REQUESTED** [review]

File: corpus/templates/registration/password_reset_done.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    Done! | Reset Password
+{% endblock %}
+
+{% block content %}
+    <div class="card w-3/4 lg:w-1/3 mx-auto my-10 bg-base-200 shadow-xl">
+        <div class="card-body">
+            <h1 class="card-title text-2xl">Reset Password</h1>
+            <p>
+                We've emailed you instructions for setting your password, if an account exists with the email you
+                entered. You should receive them shortly. The reset link is valid for <strong>30 minutes</strong>.
```
Comment (nishant-nayak) [inline corpus/templates/registration/password_reset_done.html:13]: 30 minutes is way too less for the onboarding right? Do you think we can keep it higher (maybe 24 hours) until the onboarding is done?

Once onboarding is finished I think 30 minutes is a reasonable duration for validity.

**Review by nishant-nayak — CHANGES_REQUESTED** [review]

**Comment by anirudhprabhakaran3** [comment]
Backlog issue is tracked here: https://github.com/IEEE-NITK/corpus/issues/140

**Review by nishant-nayak — APPROVED** [review]
LGTM!

**Pull request merged**